spec/PLUGIN_spec.rb

```diff
@@ -9,6 +9,34 @@ module Danger
     #
     # You should test your custom attributes and methods here
     #
+    describe 'with Dangerfile' do
+      before do
+        @dangerfile = testing_dangerfile
+        @my_plugin = testing_dangerfile.my_plugin
+      end
+
+      # Some examples for writing tests
+      # You should replace these with your own.
+
+      it "Warns on a monday" do
+        monday_date = Date.parse("2016-07-11")
+        allow(Date).to receive(:today).and_return monday_date
+
+        @my_plugin.warn_on_mondays
+
+        expect(@dangerfile.status_report[:warnings]).to eq(["Trying to merge code on a Monday"])
+      end
+
+      it "Does nothing on a tuesday" do
+        monday_date = Date.parse("2016-07-12")
+        allow(Date).to receive(:today).and_return monday_date
+
+        @my_plugin.warn_on_mondays
+
+        expect(@dangerfile.status_report[:warnings]).to eq([])
+      end
```

Comment:
Some non-compiling examples, should be enough o understand how to run code in a Danger context